drivers/interrupt_controller/intc_nxp_irqsteer.c

```diff
@@ -0,0 +1,522 @@
+			 struct irqsteer_dispatcher *dispatcher)
+{
+	int i, idx;
+
+	idx = irq;
+
+	for (i = dispatcher->master_index - 1; i >= 0; i--) {
+		idx -= IRQSTEER_GetMasterIrqCount((IRQSTEER_Type *)regmap, i);
+	}
+
+	return irq_to_level_2(idx) | dispatcher->irq;
+}
+
+static int from_zephyr_irq(uint32_t regmap, uint32_t irq, uint32_t master_index)
+{
+	int i, idx;
+
+	idx = irq;
+
+	for (i = 0; i < master_index; i++) {
+		idx += IRQSTEER_GetMasterIrqCount((IRQSTEER_Type *)regmap, i);
+	}
+
+	return idx;
+}
+
+void z_soc_irq_enable_disable(uint32_t irq, bool enable)
+{
+	uint32_t parent_irq;
+	int i, system_irq, level2_irq;
+	const struct irqsteer_config *cfg;
+
+	if (irq_get_level(irq) == 1) {
+		/* LEVEL 1 interrupts are DSP direct */
+		z_xtensa_irq_enable(XTENSA_IRQ_NUMBER(irq));
+		return;
+	}
+
+	parent_irq = irq_parent_level_2(irq);
+	level2_irq = irq_from_level_2(irq);
+
+	/* find dispatcher responsible for this interrupt */
+	for (i = 0; i < ARRAY_SIZE(dispatchers); i++) {
+		if (dispatchers[i].irq == parent_irq) {
+			cfg = dispatchers[i].dev->config;
+
+			system_irq = from_zephyr_irq(cfg->regmap_phys,
+						     level2_irq,
+						     dispatchers[i].master_index);
+
+			if (enable) {
+				IRQSTEER_EnableInterrupt((IRQSTEER_Type *)cfg->regmap_phys,
+							 system_irq);
+			} else {
+				IRQSTEER_DisableInterrupt((IRQSTEER_Type *)cfg->regmap_phys,
+							  system_irq);
+			}
+
+			break;
+		}
+	}
+
+	__ASSERT(false, "INTID %d doesn't belong to any dispatcher", irq);
+}
+
+void z_soc_irq_enable(uint32_t irq)
+{
+	z_soc_irq_enable_disable(irq, true);
+}
+
+void z_soc_irq_disable(uint32_t irq)
+{
+	z_soc_irq_enable_disable(irq, false);
+}
+
+int z_soc_irq_is_enabled(unsigned int irq)
+{
+	uint32_t parent_irq;
+	int i, system_irq, level2_irq;
+	const struct irqsteer_config *cfg;
+
+	if (irq_get_level(irq) == 1) {
+		/* LEVEL 1 interrupts are DSP direct */
+		return z_xtensa_irq_is_enabled(XTENSA_IRQ_NUMBER(irq));
+	}
+
+	parent_irq = irq_parent_level_2(irq);
+	level2_irq = irq_from_level_2(irq);
+
+	/* find dispatcher responsible for this interrupt */
+	for (i = 0; i < ARRAY_SIZE(dispatchers); i++) {
+		if (dispatchers[i].irq == parent_irq) {
+			cfg = dispatchers[i].dev->config;
+
+			system_irq = from_zephyr_irq(cfg->regmap_phys,
+						     level2_irq,
+						     dispatchers[i].master_index);
+
+			return IRQSTEER_InterruptIsEnabled((IRQSTEER_Type *)cfg->regmap_phys,
+							   system_irq);
+		}
+	}
+
+	__ASSERT(false, "INTID %d doesn't belong to any dispatcher", irq);
+
+	return false;
+}
+
+
+static void irqsteer_isr_dispatcher(const void *data)
+{
+	struct irqsteer_dispatcher *dispatcher;
+	const struct irqsteer_config *cfg;
+	uint32_t irq_num, table_idx;
+	int i, counter;
+	int system_irq, zephyr_irq;
+
+	dispatcher = (struct irqsteer_dispatcher *)data;
+	cfg = dispatcher->dev->config;
+	irq_num = IRQSTEER_GetMasterIrqCount((IRQSTEER_Type *)cfg->regmap_phys,
+					     dispatcher->master_index);
+	i = 0;
```

Comment:
You can initialize `i` and `counter` on declaration, above.

Reply:
Ah, this is more of a personal preference I guess. I want to keep the declaration separate from the initialization.